File: codes/CF1611F.cpp
#include <cstdio>
#include <vector>
#include <iostream>
#include <fstream>

using namespace std;

void CF1611F(istream& cin, ostream& cout) {
	int t;
	for(cin >> t; t>0; t--) {
		int n, s;
		cin >> n >> s;
		vector<int> nums(n);
		long long sum = 0;
		int left=0;
		int l=-1, r=-2;
		for(int i=0; i<n; i++) {
			cin >> nums[i];
			sum += nums[i];
			while(sum+s<0 && left<=i) {
				sum -= nums[left];
				left++;
			}

			if (sum+s >= 0 && r-l < i-left) {
				l = left+1;
				r = i+1;
			}
		}
		if(l==-1) {
			cout << l << endl;
		} else {
			cout << l << ' ' << r << endl;
		}
	}
	
}

int main() {
	std::ios::sync_with_stdio(false);
	std::cin.tie(nullptr);
	// ifstream infile("src/input.txt");
	// cin.rdbuf(infile.rdbuf());

    CF1611F(cin, cout);
	return 0;
}

File: codes/CF1611F.go
package main

import (
	"bufio"
	"fmt"
	"io"
	"os"
)

func CF1611F(read io.Reader, write io.Writer) {
	in := bufio.NewReader(read)
	out := bufio.NewWriter(write)
	defer out.Flush()

	var t int
	for fmt.Fscan(in, &t); t > 0; t-- {
		var n int
		var s int64
		fmt.Fscan(in, &n, &s)
		nums := make([]int, n)
		var sum int64
		left := 0
		l, r := -1, -2
		for i := 0; i < n; i++ {
			fmt.Fscan(in, &nums[i])
			sum += int64(nums[i])
			for sum+s < 0 && left <= i {
				sum -= int64(nums[left])
				left++
			}
			if sum+s >= 0 && r-l < i-left {
				l, r = left+1, i+1
			}
		}
		if l == -1 {
			fmt.Fprintln(out, l)
		} else {
			fmt.Fprintln(out, l, r)
		}
	}
}

func main() {
	// in, _ := os.Open("input.txt")
	in := os.Stdin
	CF1611F(in, os.Stdout)
}
